sources/osgShader/ShaderGenerator.js

```diff
@@ -105,18 +107,17 @@ define( [
                     continue;
 
                 if ( attr.getHash ) {
-                    hash += attr.getHash();
+                    this.pushToHashStack( attr.getHash() );
                 } else {
-                    hash += attr.getType();
+                    this.pushToHashStack( attr.hashComputeCode( attr.getType() ) );
```

Comment:
indeed.
Yet I never ever encountered something going in this branch, and don't understand why it's there.